### PR TITLE
Add sorting function for units, reimplement viz, organize code for Pkg add

### PR DESCRIPTION
**Pull request by ahwillia**
You should be able to install this now with `]` -> `add https://github.com/degleris1/CMF.jl/`

I made some other changes:

* Synthetic data is now generated from a probabilistic model using `Distributions`.
* Added REQUIRE file
* `sortperm` -> sorts units lexographically to reveal sequences.

Sorry for building so much into one PR. Let me know what you think.

### CODE REVIEW

File: examples/synthetic_data.jl [hunk 1]
```diff
@@ -0,0 +1,36 @@
+using Revise
+import PyPlot; const plt = PyPlot
+import CMF; const cmf = CMF
```
Comment (degleris1) [inline examples/synthetic_data.jl:3]: Is this type of things standard practice in Julia?

File: src/model.jl [hunk 2]
```diff
@@ -1,17 +1,12 @@
...
-include("./hals.jl")  # HALSUpdate
-include("./anls.jl") 
-include("./common.jl")
-
-
 ALGORITHMS = Dict(
     :mult => MULT,
     :hals => HALS,
-    :anls => ANLS
+    # :anls => ANLS
```
Comment (degleris1) [inline src/model.jl]: Any reason this is commented out?

**Comment by bantin** [comment]
@degleris1 @ahwillia This has a merge conflict because of Anthony's change to return the true factors. I'm going to resolve the conflicts and add a commit to this PR before we merge.

File: src/model.jl (hunk 2 above)
Comment (bantin) [reply src/model.jl]: Resolved by 8667bc4

**Comment by ahwillia** [comment]
Thanks!

**Pull request merged**